architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,296 @@
+5. UCP gets SQL Server DB creation request from Deployment Engine 
+6. UCP forwards the request to create SQL server DB to Applications RP
+7. Applications RP communicates with OCI registry, downloads the bicep recipe for creating SQLServer DB. 
+8.  RP then sends a request to UCP for deploying the bicep recipe and also stores information about it in its datastore.
+9.  UCP requests DE to deploy the SQL Server DB.
+
+
+Below is a high level overview of various key subcomponents in Applications RP
+![Applications RP](2024-10-applications-rp-threat-model/apprp.png) 
+
+
+### Implementation Details
+
+#### Use of Cryptography
+
+
+1. **Computing the Hash for Terraform Backend Configuration**: [Link to code](https://github.com/radius-project/radius/blob/main/pkg/recipes/terraform/config/backends/kubernetes.go#L110).
+
+   1. **Purpose**: The purpose of computing the hash is to compute a key for kubernetes secret which has the etrraform backend config
+   2. **Library**: The library used to calculate the hash of the deployment configuration is the crypto library, which is one of the standard libraries of Go: [Link to library](https://pkg.go.dev/crypto@go1.23.1).
+   3. **Type**: [SHA1](https://www.rfc-editor.org/rfc/rfc3174.html). Note: "SHA-1 is cryptographically broken and should not be used for secure applications." [Link to warning](https://pkg.go.dev/crypto/sha1@go1.23.1). This is used as an optimization for detecting changes, not as a security protection.
+
+2. Applications RP also uses UCP's queue and store libraries. These libraries use Cryptography and are detailed in UCP threat model. [link to be added]
+
+#### Storage of secrets
+
+Applications RP has access to sensitive information related to the application resource it manages as well as the cloud credentials it requires for managing cloud resources on Azure and AWS. 
+
+##### Managing cloud credentails
+
+Applications RP requires AWS and Azure credentials for accessing and managing resources in cloud. It fetches credentials using UCP Secret Provider library. 
+Credentials are not available for retrieval through API. The RP also supports [federated identity](https://docs.radapp.io/guides/operations/providers/overview) for both Azure and AWS. Unless there is a limitation that prevents using federated identity, users should prefer using this since it removes the need to store secrets. 
+
+Using these credentials, Applications RP can create and manage other resources in AWS and Azure. 
+
+#### Managing secrets for applications
+
+Applications RP provides a secret store which can be used to store sensitive information such as TLS certificate and private keys. It uses kubernetes secrets to implement this secret store. In most cases, sensitive application data can be stored in this secret store and consumed by the application's other components such as a container. 
```

Comment:
Is it just Kubernetes secrets implementation that is available?

Reply:
At present, yes.